cloudconfig/containerinit/container_userdata_test.go

```diff
@@ -84,80 +96,181 @@ func (s *UserDataSuite) SetUpTest(c *gc.C) {
 		ConfigType:    network.ConfigManual,
 		NoAutoStart:   true,
 	}}
-	s.expectedSampleConfig = `
-auto any0 any1 any3 lo
-
-iface lo inet loopback
-  dns-nameservers ns1.invalid ns2.invalid
-  dns-search bar foo
 
-iface any0 inet static
-  address 0.1.2.3/24
-  gateway 0.1.2.1
-
-iface any1 inet static
-  address 0.2.2.4/24
-  post-up ip route add 0.5.6.0/24 via 0.2.2.1 metric 50
-  pre-down ip route del 0.5.6.0/24 via 0.2.2.1 metric 50
-
-iface any2 inet dhcp
-
-iface any3 inet dhcp
+	for _, version := range []string{
+		"/usr/bin/python2",
+		"/usr/bin/python3",
+		"/usr/bin/python",
+	} {
+		if _, err := os.Stat(version); err == nil {
+			s.pythonVersions = append(s.pythonVersions, version)
+		}
+	}
```

Comment:
we should add an assert that len(version) >= 1 so that we know we are at least testing *something* regularly.
We don't want to require all python versions on everyone's machine, but if they are running the test suite, it shouldn't silently skip.